app/models/lookup_value.rb

```diff
@@ -75,7 +75,8 @@ def ensure_fqdn_exists
     md = ensure_matcher(/fqdn=(.*)/)
     return md if md == true || md == false
     fqdn = md[1].split(LookupKey::KEY_DELM)[0]
-    return true if Host.unscoped.find_by_name(fqdn) || host_or_hostgroup.try(:new_record?) ||
+    fqdn_matched = Host.search_for("facts.fqdn=#{fqdn}").any? || Host.unscoped.where(:name => fqdn).any?
```

Comment:
If users prefer to use short names, shouldn't the matchers also use short names?

If we keep this change, I think the order should be the other way around, first we should search the host by fqdn and then fallback to facts. I see two reasons for this, first that the host name would have higher priority in case these two values get out of sync so it would prefer current behavior, second is that the search for is more resource consuming while the host name will be mostly used.

Also shouldn't the search_for be called on `Host.unscoped`?